src/modular/dyn_residue.rs

```diff
@@ -35,13 +35,19 @@ pub struct DynResidueParams<const LIMBS: usize> {
 impl<const LIMBS: usize> DynResidueParams<LIMBS> {
     /// Instantiates a new set of `ResidueParams` representing the given `modulus` if it is odd.
     ///
-    /// Returns a `CtOption` that is `None` if the provided modulus is not odd.
-    pub fn new(modulus: &Uint<LIMBS>) -> CtOption<Self> {
-        let r = Uint::MAX.const_rem(modulus).0.wrapping_add(&Uint::ONE);
-        let r2 = Uint::const_rem_wide(r.square_wide(), modulus).0;
+    /// Returns `None` if the provided modulus is not odd.
+    pub fn new(modulus: &Uint<LIMBS>) -> Option<Self> {
```

Comment:
Why no `CtOption` here?

Reply:
I didn't feel like `CtOption` added any security there, because what are you going to do with the `None` value besides returning an error? It doesn't really make sense to chain the result into anything. And with `BoxedUint`, you actually cannot chain it because of that `subtle` problem.

But I can change it back to `CtOption` for the sake of uniformity.

Reply:
Rolled back

Reply:
Note I don't actually have any use cases which require a secret modulus, so perhaps it's not worth providing.

The reason you'd use `CtOption` in a hypothetical protocol which required one would be to follow it up with `map` and perform some subsequent operation before handling the error.

Reply:
You can do it with an `Option`, if you're going to return an error anyway, you don't need constant-timeness. But both boxed and stack residues should have the same API, so I don't mind leaving this return type alone for now.

Reply:
You might handle the error via a rejection symbol, but again, this is spitballing about a problem I don't actually have (i.e. secret modulus)

Reply:
I do use secret moduli (in some helper methods related to Paillier encryption), but they are guaranteed to be odd, so no information is revealed this way (and if it was, it's only one bit anyway)

Reply:
In that case it might be good to provide separate `new` and `new_vartime` constructors like `BoxedResidueParams`, so they can accordingly call constant-time versus variable-time remainder